srml/council/src/lib.rs

```diff
@@ -14,7 +14,186 @@
+//!
+//! - Creation of council proposals by councillors using the council motion mechanism.
+//! - Validation of council proposals.
+//! - Tallying votes of council proposals by councillors during the proposals' voting period.
+//! - Vetoing (postponement) of council proposals for a cooldown period through abstention by councillors.
+//! - Elevation of council proposals to start an associated referenda on the table of referenda.
+//! - Applying vote thresholds to referenda depending on their associated council proposal voting approval tally.
+//! - Instant autonomous execution of referenda once their vote tally reaches the vote threshold level of approval.
+//! - Cancellation of council proposals that were elevated as associated referenda onto the table of referenda.
+//! - Candidate registration in list slots (necessary to receive candidate approval votes during the
+//! presentation period).
+//! - Deduction of security bonds from candidate voters.
+//! - Express council seat candidate approval voting.
+//! - Reaping of candidate voters due to valid claims of their inactivity by reporters.
+//! - Reaping reporters that lodge invalid claims of candidate voter inactivity.
+//!
+//! ## Interface
+//!
+//! ### Dispatchable Functions
+//!
+//! #### Council Motions
+//!
+//! - `propose` - Create a council proposal using the council motion mechanism by a councillor who provides a vote
+//! threshold.
+//! - `vote` - Update the approval vote tally through active councillors voting on a council proposal. Approve and
+//! execute it when the vote tally reaches the vote threshold. Disapprove it when the potential votes remaining is
+//! less than the threshold.
+//!
+//! #### Council Seats
+//!
+//! - `set_approvals` - Set candidate approvals. Approval slots stay valid as long as candidates in those slots are
+//! registered.
+//! - `proxy_set_approvals` - Set candidate approvals from a proxy. Approval slots stay valid as long as candidates
+//! in those slots are registered.
+//! - `reap_inactive_voter` - Remove a voter. Can be called by anyone. Returns the voter deposit to the `signed`
+//! origin.
+//! - `retract_voter` - Remove a voter. All votes are cancelled and the voter deposit is returned.
+//! - `submit_candidacy` - Submit oneself for candidacy. Account must have enough transferrable funds in it to pay
+//! the bond.
+//! - `present_winner` - Claim that the `signed` origin is one of the top candidates.
+//! - `set_desired_seats` - Set the desired council member count.
+//! - `remove_member` - Remove a council member immediately. A tally happens instantly (if not already in a
+//! presentation period) to fill the seat if removal means that the desired members are not met.
+//! - `set_presentation_duration` - Set the presentation duration.
+//! - `set_term_duration` - Set the term duration.
+//! - `on_finalize` - Signature declaration that runs anything that needs to be done at the end of the block.
```

Comment:
`finalize_block` is not really council-related per se.

Reply:
thanks i've removed it. it says what `finalize_block` is for in paritytech/substrate/core/sr-primitives/src/traits.rs anyway